### ISSUE TITLE
Command to rollback to a saved git head id

### ISSUE THREAD

**Issue by Dougie777** [paul-gauthier/aider#221]
After using this for more than 8 hours a day for the past week, I can say that although it is very powerful, it would be immensely useful to be able to rollback to a key point within aider.  So when I try to do a new thing, (maybe tie it to the /clear command) I want aider to take note of the current git commit id so that later I could rollback all the changes to this point in case everything gets mucked up.  Although its a no brainer to save this in /clear it should also have a /SavePoint or something like that.  Then I could do a rollback to that point somehow.  Anyway its an awesome tool but I find I can go down paths I need to get back from and finding the correct spot in git is very hard because the commit descriptions are hard to follow (for me anyway)

**Comment by paul-gauthier** [comment]
Thanks for trying aider! Glad to hear you are getting so much use from it!

Git tags might be a nice way to achieve what you'd like. You could do `/git tag <checkpoint-name>` whenever you want to make a checkpoint for yourself. Then if you want to roll back to that checkpoint you could do `/git reset --hard <checkpoint-name>`.

Be aware though that git is a powerful tool with many sharp edges. So make sure you understand the git commands you're using, especially `reset`.

**Comment by ryanfreckleton** [comment]
Hello all,

I've also been enjoying using Aider, and have a few tricks to solve similar situations:

1. You can change the description of your current git commit with `git commit --amend`. This is a good idea to do before you do deep explorations so you can see and revert to a known good git. You can also do `git commit --amend -m "<commit message>"` if you wish to do it in one line.
2. I often will work with two or more branches, one that is `spike`, where experiments are going that may or may not succeed, as well as `dev` where changes that will eventually go into a merge/pull request live. If the `spike` turns out not to be what I want or need, I can just reset / force it back to `dev`.

**Comment by paul-gauthier** [comment]
I'm going to close this issue for now, but feel free to re-open or file a new issue any time.
